### PR TITLE
move integration tests out of main dub.json

### PR DESCRIPTION
**Pull request by John-Colvin**
which avoids dub conflicts with mir versions.

### CODE REVIEW

**Comment by John-Colvin** [comment]
The tests are failing, but completely randomly including what looks like dub segfaulting in some cases.

It all works fine for me locally.

**Comment by John-Colvin** [comment]
Done some more local tests with recent dmd and ldc versions, it all seems to work.

File: .travis.yml [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
   include:
     - d: dmd-nightly
-      script: dub run --root mir-integration && dub run --build=unittest -c integration-mir
-    - d: dmd-2.087.0
-    - d: dmd-2.086.1
-    - d: dmd-2.085.1
-    - d: ldc-1.16.0
-    - d: ldc-1.15.0
-    - d: ldc-1.14.0
+    - d: dmd
```
Comment (atilaneves) [inline .travis.yml]: Adding `dmd` and `ldc` without versions and not allowing failures for them makes the build not be deterministic.

Comment (John-Colvin) [reply .travis.yml]: Added them to allowed failures

**Pull request merged**